CDCgov/prime-simplereport
# Parallelize queue successful message deletion

# Azure Function PULL REQUEST

## Related Issue

- fixes #4147

## Changes Proposed

- Parallelize the deletion of the successful message from the queue
- Update dependencies to fix failing tests

## Additional Information

- hardcoded `node-fetch` to  `2.6.6` since the lastest 2.6.7 (as of right now) is causing the following import error
`node_fetch_1.Headers is not a constructor`
- considered upgrading to `node-fetch v3`, but thats a wsi only package and it seems like it might need a of work to migrate since its not compatible with `CommonJS`. perhaps we should move to axios if node-fetch v2 stops being supported.
- added `package-lock.json` to lock the dependencies
- tested on dev

## Testing

- How should reviewers verify this PR?
- deploy branch to any dev env
- login to azure, submit some tests, verify records in the queue, observe the function app logs and trace when the records get removed

## Checklist for Primary Reviewer
